### ISSUE TITLE
feature ideas

### ISSUE THREAD

**Issue by datapimp** [mikeyamadeo/prepacked#1]
This project is shaping up to provide really solid UX benefits for webpack.  Great work, I would love to help.

Question:

How would I go about using `prepacked` as a single dependency that bootstrapped a webpack config with HMR and whatever set of babel, eslint, react dependencies I specified in a `.prepackedrc` file?  (Even going as far as Redux, React-Bootstrap, and whatever else).

I imagine it could behave much like babel-presets.

In a development / prototyping context it would would be really nice to be able to run `prepacked ./path/to/entry` and not even need to include any configuration or dependencies.  I have had limited success with this -- i've gotten it to work but can't help but thinking I am making it too difficult, and it definitely doesn't have as enjoyable of a UX as `prepacked` does already.

**Comment by mikeyamadeo** [comment]
This is great! Thanks for taking time to lend your grey matter to this. 

So there is a tricky balance that needs to be determined. i.e. the balance of convention / opinion vs configuration / extensibility.

From my perspective, I think the current suite of tools available has done an excellent job with providing configuration & extensibility. This also happens to be contributing to the configuration / tooling fatigue I believe we are trying to minimize. With a significant motivation for `prepacked` being to get up and running with little to no configuration, I wrestle with the idea of solving configuration fatigue with a additional configuration via a `.prepackedrc` for example.

I like the approach taken by [standard.js](http://standardjs.com/):
"No decisions to make. No .eslintrc, .jshintrc, or .jscsrc files to manage. It just works."
There is a core set of best practices here that people just kind of accept and use. My personal adoption of this has eliminated any need for eslint configuration and I like that I have fewer decision on my plate as a result. With that said, it's also extensible for more advanced users to add custom rules _in addition_ to the core conventions.

In short, new users don't have to think about configuration and more advanced users have the flexibility to extend it. That's kind of where my head is at with this project.

You may be saying as much above in suggesting a `.prepackedrc` as a mode of extension. I wonder if we could get away with all the extensibility configuration we need within `package.json` itself? As for simply getting up and going with the "prepacked" conventions, I really like your idea of boiling it down into a command line script. Can you elaborate on what you've had "limited success with"?

Thanks!

**Comment by datapimp** [comment]
I think that constraining ourselves to only a key within the `package.json` is a great idea.  No new files, no configuration required -- it just works.  That's what I love.

I only want something as a means of extensibility -- and really I just want the ability to switch between a few different presets, which is something we could still curate.

The one size fits all approach just doesn't work right now for me and I don't see any way it could.  However, 2-3 presets would allow me to have prepacked take over a ton of build / test / dev acrobatics across a number of different projects / teams etc.

**Comment by mikeyamadeo** [comment]
Great. I think we are on the same page with extensibility over additional configuration. Babel-like presets within `package.json` feels like a clean start to me. Let's use your needs as a starting point to discuss what presets to support. What did you have in mind?
